Fix the Privacy Policy help notice

desrosj opened the pull request:
Related issue: #10448.
Depends on #11604.

The new editor does not support the `edit_form_after_title` action hook. Because WordPress Core uses this hook to output the notice, it is not printed to the screen.

After #11604 is merged, legacy style admin notices (`<div class="notice">...notice content...</div>`) will be consumed by the Notices API and displayed. This change ensures that when #11604 is merged the privacy policy notice will appear again when editing the privacy policy page.

This PR moves the notice to the `admin_notices` action hook to ensure it is printed to the page and is consumable.

To test, ensure the `Need help putting together your new Privacy Policy page?` notice is output in the page source only when editing the privacy policy page. To further test, pull this change into the `update/legacy-notices` branch locally, and verify the notice correctly appears when editing the privacy policy page.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style. <!-- Check code: `npm run lint`, Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/javascript/ -->
- [ ] ~My code follows the accessibility standards.~ (NA) <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/coding-standards/accessibility-coding-standards/ -->
- [x] My code has proper inline documentation. <!-- Guidelines: https://make.wordpress.org/core/handbook/best-practices/inline-documentation-standards/javascript/ -->

desrosj committed:
Display the privacy policy help notice with the `admin_notices` action hook.

The new editor does not support the `edit_form_after_title` action hook. Because WordPress Core uses this hook to output the notice, it is not printed to the screen.

After #11604 is merged, legacy style admin notices (`<div class="notice">...notice content...</div>`) will be consumed by the Notices API and displayed. This change ensures that when #11604 is merged the privacy policy notice will appear again when editing the privacy policy page.

desrosj committed:
Use `get_post()` instead of `global $post`.

aduth commented:
Is there an associated Trac ticket for which this can be considered to be handled more gracefully internal to core itself, as part of the Gutenberg merge?

desrosj committed:
Add missing `@since` documentation tag.

desrosj committed:
Only display the notice if the function is actually hooked to `edit_form_after_title`.

This prevents the notice from displaying twice if the notice has already been moved to the `admin_notices` hook (as would happen in https://core.trac.wordpress.org/attachment/ticket/45057/45057.diff).

gziolo commented:
So this PR depends on #11604. On its own it doesn't display the notice when I edit Privacy Policy page.

gziolo commented:
I tested it with #11604 and it works perfectly fine:

![screen shot 2018-11-19 at 11 10 13](https://user-images.githubusercontent.com/699132/48700211-c0d7a500-ebeb-11e8-8439-dfdae728d91f.png).

Let's merge it and make sure that it still works properly when #11604 lands.
